src/utils/form.tsx

```diff
@@ -126,6 +126,20 @@ function hasRegisteredFormComponentParent(
   return hasRegisteredFormComponentParent;
 }
 
+/**
+ * Helper to submit a form.
+ */
+export function submitForm(formEl: HTMLFormElement): void {
```

Comment:
How about passing the form component and we can grab the `formEl` from there? It'd be consistent with existing utils here (e.g., `connectForm`).

Reply:
The problem with that is it won't work for the button case since the button isn't a FormElement, it just submits it :\

Reply:
Gotcha. Let's leave button out of this then since it's already doing its own thing and the original issue is about form components.

Reply:
I think its better now 🎉

Reply:
Oooh, yusss. 🤩

Last feedback on the draft is maybe renaming `FormSubmitComponent` to `FormOwner` and adding some internal doc on it for guidance.